static/app/utils/useResolveRoute.tsx

```diff
@@ -7,16 +7,37 @@ import {OrganizationContext} from 'sentry/views/organizationContext';
 import shouldUseLegacyRoute from './shouldUseLegacyRoute';
 import {normalizeUrl} from './withDomainRequired';
 
+/**
+ * In yarn dev-ui mode we proxy API calls to sentry.io.
+ * However, all of the browser URLs are either acme.localhost,
+ * or acme.dev.getsentry.net so we need to hack up the server provided
+ * domain values.
+ */
+function localizeDomain(domain?: string) {
+  if (!window.__SENTRY_DEV_UI || !domain) {
+    return domain;
+  }
+  const host = window.location.host;
+  const validHostPattern =
+    /((?:localhost|[^.]+\.sentry\.dev|dev\.getsentry\.net)(?:\:\d*)?)/;
```

Comment:
Should this be some kind of constant, we have a similar regex above

Reply:
The capturing groups are different between the patterns though. The one in `useResolveRoute()` needs to extract the slug, while this function needs to match the trailing domain + port.

I can try replacing the regex patterns with split on `.` operations.